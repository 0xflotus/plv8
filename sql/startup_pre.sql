CREATE TABLE public.plv8_modules (
   modname name primary key,
   code    text not null
);


insert into plv8_modules values ('testme','bar=98765;');

create function startup()
   returns void
   language plv8 as 
$$

foo=14378;

load_module = function(modname) 
{
    var rows = executeSql("SELECT code from plv8_modules where modname = $1", [modname]);
    for (var r = 0; r < rows.length; r++)
    {
        var code = rows[r].code;
        eval("(function() { " + code + "})")();
        print (NOTICE, 'loaded module: ' + modname);
    }
};

$$;
