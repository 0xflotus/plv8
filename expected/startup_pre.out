CREATE TABLE public.plv8_modules (
   modname name primary key,
   code    text not null
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "plv8_modules_pkey" for table "plv8_modules"
insert into plv8_modules values ('testme','bar=98765;');
create function startup()
   returns void
   language plv8 as 
$$

foo=14378;

load_module = function(modname) 
{
    var rows = executeSql("SELECT code from plv8_modules where modname = $1", [modname]);
    for (var r = 0; r < rows.length; r++)
    {
        var code = rows[r].code;
        eval("(function() { " + code + "})")();
        print (NOTICE, 'loaded module: ' + modname);
    }
};

$$;
